### ISSUE TITLE
Improve @require download checks by supporting SRI hashes

### ISSUE THREAD

**Issue by stdedos** [Tampermonkey/tampermonkey#1291]
Improve https://github.com/Tampermonkey/tampermonkey/issues/255 by supporting base64-encoded SRIs 

## Expected Behavior

Support https://code.jquery.com/jquery-3.4.1.min.js#sha256-WpOohJOqMqqyKL9FccASB9O0KwACQJpFTUBLTYOVvVU= URLs also

## Actual Behavior

Only https://code.jquery.com/jquery-3.4.1.min.js#sha256=5a93a88493aa32aab228bf4571c01207d3b42b0002409a454d404b4d8395bd55 is valid.

Because of the different hash representations, and different artifact versions (original vs minified), I ended up comparing original `WpOohJOqMqqyKL9FccASB9O0KwACQJpFTUBLTYOVvVU=` (i.e. `0925e8ad7bd971391a8b1e98be8e87a6971919eb5b60c196485941c3c1df089a`) vs the minified `5a93a88493aa32aab228bf4571c01207d3b42b0002409a454d404b4d8395bd55` (i.e. `CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=`)

If there was no need for conversion, I would've simply copied the original values from the site

## Specifications

- Chromium: (i.e. 64.0.3282.140)
- TM: (i.e. 4.5.5648)
- OS: (i.e. MacOS 10.13.4)

**Comment by derjanb** [comment]
`https://code.jquery.com/jquery-3.4.1.min.js`'s SHA-256 sum is `0925e8ad7bd971391a8b1e98be8e87a6971919eb5b60c196485941c3c1df089a`

```sh
wget https://code.jquery.com/jquery-3.4.1.min.js
sha256sum jquery-3.4.1.min.js
```

prints

```
0925e8ad7bd971391a8b1e98be8e87a6971919eb5b60c196485941c3c1df089a  jquery-3.4.1.min.js
```


`WpOohJOqMqqyKL9FccASB9O0KwACQJpFTUBLTYOVvVU=` now decodes to `"5a93a88493aa32aab228bf4571c01207d3b42b0002409a454d404b4d8395bd55"` which is not the right SHA-256 sum for this file. So `// @require https://code.jquery.com/jquery-3.4.1.min.js#sha256=5a93a88493aa32aab228bf4571c01207d3b42b0002409a454d404b4d8395bd55` is working neither.

The working links are:

```JS
// @require https://code.jquery.com/jquery-3.4.1.min.js#sha256-0925e8ad7bd971391a8b1e98be8e87a6971919eb5b60c196485941c3c1df089a
// @require https://code.jquery.com/jquery-3.4.1.min.js#sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=
```

which are both working fine. 

Also copying the values from https://code.jquery.com/ is working fine here:

![image](https://user-images.githubusercontent.com/767504/126270044-81e0fb8d-b3c3-4138-b3fd-caf43e67d5e3.png)

```JS
// @require https://code.jquery.com/jquery-3.6.0.min.js#sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=
```

Do I miss something?

**Comment by derjanb** [comment]
> I didn't want to do that, because it defeats the purpose of checking the hash: if I use my own, then, what good is it to check it? confused

The point was that the hashes you mentioned in https://github.com/Tampermonkey/tampermonkey/issues/1291#issue-947928004 (5a93a... and CSXor...) don't match the link's content, which is why Tampermonkey refuses to load the `@require`.

So from my point of view everything is working as expected. :sweat_smile:

**Comment by stdedos** [comment]
> The point was that the hashes you mentioned in [#1291 (comment)](https://github.com/Tampermonkey/tampermonkey/issues/1291#issue-947928004) (5a93a... and WpOo...) don't match the link's content, which is why Tampermonkey refuses to load the `@require`.

> > Because of the different hash representations, and different artifact versions (original vs minified), I ...

tl;dr I goofed 😅 

> So from my point of view everything is working as expected. 😅

It might be nice to diversify your examples (e.g. in the pages above), if not emphasize a tad more that base64 SRIs are valid - but yeah, otherwise "works as designed"

**Comment by derjanb** [comment]
Done. :+1:

**Comment by stdedos** [comment]
Thank you 😄
